app/code/Magento/Customer/Model/AccountManagement.php

```diff
@@ -855,9 +864,33 @@ public function createAccount(CustomerInterface $customer, $password = null, $re
         } else {
             $hash = null;
         }
+        //If Customer group id specified in the request, Check the group id exist in Magento.
+        if ($customer->getGroupId()) {
+            if (!$this->checkCustomerGroupExist($customer->getGroupId())) {
+                throw new LocalizedException(
+                    __("The group id specified is invalid. Please specify valid group id.")
+                );
+            }
+        }
         return $this->createAccountWithPasswordHash($customer, $hash, $redirectUrl);
     }
 
+    /**
+     * Check whether the group id exist or not.
+     *
+     * @param int $groupId
+     * @return int
+     * @throws NoSuchEntityException
+     */
+    protected function checkCustomerGroupExist($groupId)
```

Comment:
Please, use the `private scope` for this method and add the corresponding type hinting

```suggestion
    private function checkCustomerGroupExist(int $groupId): bool
```

Reply:
@rogyar method scope changed and type hints added. Thanks for this suggestion. I will use type hint in my future contribution without a failure.